### PR TITLE
[jsk_perception] support image with alpha in image_publisher

### PR DESCRIPTION
**Pull request by knorth55**
support image with alpha such as `rgba8`, `bgra8` in `image_publisher`

### CODE REVIEW

File: jsk_perception/node_scripts/image_publisher.py [hunk 1]
```diff
@@ -84,30 +84,39 @@ def publish(self, event):
                 info.R = [1, 0, 0, 0, 1, 0, 0, 0, 1]
             self.pub_info.publish(info)
 
-    def cv2_to_imgmsg(self, img_bgr, encoding):
+    def cv2_to_imgmsg(self, img, encoding):
         bridge = cv_bridge.CvBridge()
         # resolve encoding
-        if getCvType(encoding) == 0:
+        if getCvType(encoding) in [0, 2, 5]:
```
Comment (k-okada) [inline jsk_perception/node_scripts/image_publisher.py]: how about `getCvType(encoding) in [cv2.CV_8UC1, cv2.CV_16UC1, cv2.CV_32FC1]`

Comment (knorth55) [reply jsk_perception/node_scripts/image_publisher.py]: Thank you, I updated.

**Comment by k-okada** [comment]
@knorth55 something wrong with your update,
```
k-okada@p51s:~/catkin_ws/ws_recognition/src/jsk_recognition$ rostest -t jsk_perception/test/matchtemplate.test 
... logging to /home/k-okada/.ros/log/rostest-p51s-10713.log
testtest_matchtemplate (rostest.runner.RosTest) ... [ROSTEST]setup[jsk_perception/test/matchtemplate.test] run_id[52f55872-95b1-11ea-bc51-00bb605aa0af] starting
started roslaunch server http://p51s:38127/

SUMMARY
========

PARAMETERS
 * /image_publisher_reference/file_name: /home/k-okada/cat...
 * /image_publisher_reference/publish_info: False
 * /image_publisher_search/file_name: /home/k-okada/cat...
 * /image_publisher_search/publish_info: False
 * /rosdistro: melodic
 * /rosversion: 1.14.5
 * /test_matchtemplate/timeout_0: 30
 * /test_matchtemplate/timeout_1: 30
 * /test_matchtemplate/timeout_2: 30
 * /test_matchtemplate/topic_0: /matchtemplate/re...
 * /test_matchtemplate/topic_1: /matchtemplate/de...
 * /test_matchtemplate/topic_2: /matchtemplate/cu...

NODES
  /
    image_publisher_reference (jsk_perception/image_publisher.py)
    image_publisher_search (jsk_perception/image_publisher.py)
    matchtemplate (jsk_perception/matchtemplate.py)
    reference_point_publisher (rostopic/rostopic)
    search_rect_publisher (rostopic/rostopic)

[ROSTEST]setup[jsk_perception/test/matchtemplate.test] run_id[52f55872-95b1-11ea-bc51-00bb605aa0af] done
[ROSTEST]Running test [test_matchtemplate]
ROS_MASTER_URI=http://p51s:38625/
setting /run_id to 52f55872-95b1-11ea-bc51-00bb605aa0af
process[image_publisher_search-1]: started with pid [10731]
process[image_publisher_reference-2]: started with pid [10732]
process[reference_point_publisher-3]: started with pid [10733]
process[search_rect_publisher-4]: started with pid [10734]
process[matchtemplate-5]: started with pid [10735]
[ROSTEST]removing previous test results file [/home/k-okada/.ros/test_results/jsk_perception/rosunit-test_matchtemplate.xml]
[ROSTEST]running test test_matchtemplate
process[test_matchtemplate-6]: started with pid [10736]
test_published (__main__.TestTopicPublished)
Test topics are published and messages come ... WARNING: topic [/matchtemplate/current_template] does not appear to be published yet
[ERROR] [1589439943.785768]: unsupported encoding: bgr8
FAIL

======================================================================
FAIL: test_published (__main__.TestTopicPublished)
Test topics are published and messages come
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/ros/melodic/share/jsk_tools/src/test_topic_published.py", line 96, in test_published
    checker.topic_name)
AssertionError: Topic [/matchtemplate/current_template] is not published

----------------------------------------------------------------------
Ran 1 test in 30.010s

FAILED (failures=1)
-------------------------------------------------------------
SUMMARY:
 * RESULT: FAIL
 * TESTS: 1
 * ERRORS: 0 []
 * FAILURES: 1 [test_published]

[test_matchtemplate-6] process has died [pid 10736, exit code 1, cmd /opt/ros/melodic/share/jsk_tools/src/test_topic_published.py --gtest_output=xml:/home/k-okada/.ros/test_results/jsk_perception/rosunit-test_matchtemplate.xml --text __name:=test_matchtemplate __log:=/home/k-okada/.ros/log/52f55872-95b1-11ea-bc51-00bb605aa0af/test_matchtemplate-6.log].
log file: /home/k-okada/.ros/log/52f55872-95b1-11ea-bc51-00bb605aa0af/test_matchtemplate-6*.log
[ROSTEST]test [test_matchtemplate] finished
[ROSTEST]retry is disabled in --text mode
[ROSTEST][ROSTEST] test [test_matchtemplate] done
[ROSTEST]tearDown[jsk_perception/test/matchtemplate.test]
shutting down processing monitor...
[matchtemplate-5] killing on exit
[search_rect_publisher-4] killing on exit
[reference_point_publisher-3] killing on exit
[image_publisher_reference-2] killing on exit
[image_publisher_search-1] killing on exit
... shutting down processing monitor complete
[ROSTEST]rostest teardown jsk_perception/test/matchtemplate.test complete
ok

----------------------------------------------------------------------
Ran 1 test in 32.849s
```

**Comment by knorth55** [comment]
@k-okada Thank you. I solved the problem.

**Pull request merged**